src/analysis/souffle/taint.dl

```diff
@@ -52,6 +52,30 @@
 // of the taint analysis).
 .decl claimNotEdge(principal: Principal, src: AccessPath, dst: AccessPath)
 
+// Indicates that an AccessPath is given an integrity tag directly. This rule
+// should only be populated with base facts with no derivation to prevent us
+// from getting tangled up in stratification issues. Propagation will happen
+// via the mayHaveInvIntegTag rule.
+.decl directHasIntegTag(accessPath: AccessPath, integTag: IntegTag)
```

Comment:
what about `ownership`?

Reply:
I didn't handle it in this PR because it wasn't necessary to show simple propagation behavior and I wanted to get some feedback on this approach first. I can add that in if you'd like, however.

Reply:
We should think about it as confidentiality labels have ownership. Also, unlike confidentiality tags, adding an integrity tag is a privileged operation.

Reply:
Added ownership aspect.